File: algorithms/codeforces/Sereja and Dima.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    int Sereja = 0;
    int Dima = 0;
    int l = 0, r = n - 1;
    while (l <= r) {
        if (a[l] > a[r]) {
            Sereja += a[l];
            l++;
        }
        else {
            Sereja += a[r];
            r--;
        }

        if (l > r) {
            break;
        }

        if (a[l] > a[r]) {
            Dima += a[l];
            l++;
        }
        else {
            Dima += a[r];
            r--;
        }
    }

    cout << Sereja << " " << Dima;

    return 0;
}

File: algorithms/python/intermediate/algorithmic_complexity/Sereja and Dima.py
n = int(input())
a = list(map(int, input().split()))

left = 0
right = n - 1
Sereja = 0
Dima = 0
turn = 0
while left <= right:
    if a[left] > a[right]:
        current = a[left]
        left += 1
    else:
        current = a[right]
        right -= 1
    
    if turn % 2:
        Dima += current
    else:
        Sereja += current
    
    turn += 1

print(Sereja, Dima)